### PR TITLE
Get weight log periods

### PR DESCRIPTION
**Pull request by pamo**
https://jira.onemedical.com/browse/RC-119

The body time series api lets us get weight data for a period of time but will automatically fill in gaps in the data with the last entered weight.

### CODE REVIEW

**Review by sophiekrisch — APPROVED** [review]
Approved, but should we add some tests?

File: lib/fitbit_api/body.rb [hunk 1]
```diff
@@ -51,5 +52,33 @@ def log_body_fat(opts)
...
+        end_date: Date.today,
+      }.merge(opts)
+
+      start_date = opts[:start_date]
+      end_date = opts[:end_date]
+      period = opts[:period]
+      resource = opts[:resource]
+
+      if period
+        result = get(endpoint_templates[:period] % { user_id: user_id, resource: resource, start_date: format_date(start_date), end_date: format_date(end_date), period: period }, opts)
```
Comment (shanear) [inline lib/fitbit_api/body.rb]: I really like the `%` used here. Did not know about that.

**Review by shanear — APPROVED** [review]
LGTM

**Comment by pamo** [comment]
> Approved, but should we add some tests?

absolutely, the original gem is light on unit tests so I may take some time to re-familiarize myself with rspec. 😂

**Review by bsy — APPROVED** [review]

**Pull request merged**